docs/sphinx/source/pvsystem.rst

```diff
@@ -0,0 +1,165 @@
+`PVSystem.module_parameters`.
+
+.. ipython:: python
+
+    module_parameters = {'pdc0': 10, 'gamma_pdc': -0.004}
+    system = pvsystem.PVSystem(module_parameters=module_parameters)
+    print(system.module_parameters)
+
+Extrinsic data is passed to a PVSystem as method arguments. For example,
+the :py:meth:`~pvlib.pvsystem.PVSystem.pvwatts_dc` method accepts extrinsic
+data irradiance and temperature.
+
+.. ipython:: python
+
+    pdc = system.pvwatts_dc(1000, 30)
+    print(pdc)
+
+Compare the :py:meth:`~pvlib.pvsystem.PVSystem.pvwatts_dc` method signature
+to the :py:func:`~pvlib.pvsystem.pvwatts_dc` function signature:
+
+    * :py:meth:`PVSystem.pvwatts_dc(g_poa_effective, temp_cell) <pvlib.pvsystem.PVSystem.pvwatts_dc>`
+    * :py:func:`pvwatts_dc(g_poa_effective, temp_cell, pdc0, gamma_pdc, temp_ref=25.) <pvlib.pvsystem.pvwatts_dc>`
+
+How does this work? The :py:meth:`~pvlib.pvsystem.PVSystem.pvwatts_dc`
+method looks in `PVSystem.module_parameters` for the `pdc0`, and
+`gamma_pdc` arguments. Then the :py:meth:`PVSystem.pvwatts_dc
+<pvlib.pvsystem.PVSystem.pvwatts_dc>` method calls the
+:py:func:`pvsystem.pvwatts_dc <pvlib.pvsystem.pvwatts_dc>` function with
+all of the arguments and returns the result to the user. Note that the
+function includes a default value for the parameter `temp_ref`. This
+default value may be overridden by specifying the `temp_ref` key in the
+`PVSystem.module_parameters` dictionary.
+
+.. ipython:: python
+
+    system.module_parameters['temp_ref'] = 0
+    # lower temp_ref should lead to lower DC power than calculated above
+    pdc = system.pvwatts_dc(1000, 30)
+    print(pdc)
+
+Multiple methods may pull data from the same attribute. For example, the
+`PVSystem.module_parameters` attribute is used by the DC model methods
+as well as the incidence angle modifier methods.
+
+
+.. _pvsystemattributes:
+
+PVSystem attributes
+-------------------
+
+Here we review the most commonly used PVSystem attributes. Please see
+the :py:class:`~pvlib.pvsystem.PVSystem` class documentation for a
+comprehensive list.
+
+The first PVSystem parameters are `surface_tilt` and `surface_azimuth`.
+These parameters are used in PVSystem methods such as
+:py:meth:`~pvlib.pvsystem.PVSystem.get_aoi` and
+:py:meth:`~pvlib.pvsystem.PVSystem.get_irradiance`.
+
+.. ipython:: python
+
+    # 20 deg tilt, south-facing
+    system = pvsystem.PVSystem(surface_tilt=20, surface_azimuth=180)
+    print(system.surface_tilt, system.surface_azimuth)
+
+    # call get_aoi with solar_zenith, solar_zenith
+    aoi = system.get_aoi(30, 180)
+    print(aoi)
+
+`module_parameters` and `inverter_parameters` contain the data
+necessary for computing DC and AC power using one of the available
+PVSystem methods. These are typically specified using data from
+the :py:func:`~pvlib.pvsystem.retrieve_sam` function:
+
+.. ipython:: python
+
+    modules = pvsystem.retrieve_sam('cecmod')
+    module_parameters = modules['Example_Module']
```

Comment:
`pvsystem.retrieve_sam` returns a `dict` with a key being a module name, and its value being a `dict` of model parameters for the module.

Reply:
Technically a dataframe, but it walks like a dict and quacks like a dict, so let's call it a dict here for simplicity. And because I think we might change it in the future.